<launch>
    <!-- Argument for node name -->
    <arg name="node_name" default="pid_twist_controller"/>

    <!-- Arguments for remapping topics -->
    <arg name="input_twist_topic" default="/input_twist"/>
    <arg name="output_twist_topic" default="/output_twist"/>

    <!-- use yaml instead of args for setting parameters -->
    <arg name="pid_values_path" default="$(find additive_manufacturing_helpers)/config/pid_twist_controller.yaml"/>
    
    <group ns="$(arg node_name)">
    <rosparam command="load" file="$(arg pid_values_path)"/>
        <!-- Node definition -->
        <node name="$(arg node_name)" pkg="additive_manufacturing_helpers" type="pid_twist_controller.py" output="screen">
            <!-- Remap topics -->
            <remap from="input_twist" to="$(arg input_twist_topic)"/>
            <remap from="output_twist" to="$(arg output_twist_topic)"/>
        </node>
    </group>
</launch>